CI fails upon touching src/test/qgistest.h

### What is the bug or the crash?

See the CI failure for PR https://github.com/qgis/QGIS/pull/51910 which only touched qgistest.h
and temptative email clarification in https://lists.osgeo.org/pipermail/qgis-developer/2023-February/065542.html

### Steps to reproduce the issue

Make an innocent modification in src/test/qgistest.h and file a pull request with that change

### Versions

Current master as of commit 3796b7b7fc9aebcf898416071b559ff4d04572cd

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

_No response_

```sh
$ scripts/clang-tidy.sh -p $PWD/build src/test/qgstest.h
19695 warnings and 1 error generated.
Error while processing /home/grayshade/Projects/QGIS/src/test/qgstest.h.
/home/grayshade/Projects/QGIS/src/test/qgstest.h:49:10: error: 'qgis_test.h' file not found [clang-diagnostic-error]
#include "qgis_test.h"
         ^~~~~~~~~~~~~
[snip]
```

--- a/.docker/docker-qgis-clangtidy.sh
+++ b/.docker/docker-qgis-clangtidy.sh
@@ -36,7 +36,11 @@ echo "::endgroup::"
 
 echo "${bold}Run clang-tidy on modifications...${endbold}"
 CLANG_TIDY_CHECKS=$(cat tests/code_layout/clangtidy_checks.txt | grep -ve "^#" | grep -ve "^$" | tr -d '\n')
-git diff -U0 HEAD^ | python3 clang-tidy-diff.py -p1 -path=${CTEST_BUILD_DIR} -use-color -checks="$CLANG_TIDY_CHECKS" | tee clang-tidy.log
+
+# We need to add build/src/test dir as extra include directories because when clang-tidy tries to process qgstest.h
+# it has no compile_commands.json instructions to know what are include directories
+# It manages to figure out for other headers though, I don't get how...
+git diff -U0 HEAD^ | python3 clang-tidy-diff.py -p1 -path=${CTEST_BUILD_DIR} -use-color -checks="$CLANG_TIDY_CHECKS" -extra-arg=-I${CTEST_BUILD_DIR}/src/test/ | tee clang-tidy.log
 
 echo -e "\e[1;34mTo reproduce locally:"
 echo -e "\e[1;34m - launch cmake with option -DCMAKE_EXPORT_COMPILE_COMMANDS=ON"

